Project/Source Files/CommitsRequest/CommitsRequestProvider.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  CommitsRequestProvider.swift
+//  NetguruCollegeApp
+//
+
+
+import Foundation
+
+protocol CommitsRequestProviderDelegate: NSObject {
+    func commitsRequestFetched(commitsRequest: [CommitsResponse])
+    func errorOccured()
+}
+
+final class CommitsRequestProvider {
+    private let apiClient: APIClientProtocol
+
+    weak var delegate: CommitsRequestProviderDelegate?
+
+    init(apiClient: APIClientProtocol) {
+        self.apiClient = apiClient
+    }
+
+    func commitsRequests(for owner: String, in repo: String) {
+        let request = CommitsRequestApiRequest(owner: owner, repo: repo)
+        apiClient.sendRequest(request: request, success: { data in
+            guard let data = data, let objects = try? CommitsResponseParser().parceResponse(data: data) else {
+                self.delegate?.errorOccured()
```

Comment:
during the workshops i have to mentioned that but, all calls to self inside the completion blocks should use `weak self` to prevent reference cycle, to do that please change line 25 to: 
`apiClient.sendRequest(request: request, success: { [weak self] data in`